tests/integration_tests/functional/test_uffd.py

```diff
@@ -100,3 +105,70 @@ def test_unbinded_socket(bin_cloner_path, test_microvm_with_api):
     assert "Load microVM snapshot error: Cannot connect to UDS in order to" \
            " send information on handling guest memory page-faults due to: " \
            "Connection refused (os error 111)" in response.text
+
+
+def test_valid_handler(bin_cloner_path,
+                       test_microvm_with_api,
+                       uffd_handler_paths):
+    """
+    Test valid uffd handler scenario.
+
+    @type: functional
+    """
+    logger = logging.getLogger("uffd_unbinded_socket")
+
+    logger.info("Create snapshot")
+    snapshot = create_snapshot(bin_cloner_path)
+
+    logger.info("Load snapshot, mem %s", snapshot.mem)
+    vm_builder = MicrovmBuilder(bin_cloner_path)
+    vm = test_microvm_with_api
+    vm.spawn()
+
+    jailed_mem = vm.create_jailed_resource(snapshot.mem)
+
+    # Copy page fault binary into chroot of microVM.
+    jailed_handler = os.path.join(vm.chroot(), "valid_handler")
+    shutil.copyfile(uffd_handler_paths['valid_handler'], jailed_handler)
+    # Change its permissions to the UID and GID of the jail.
+    cmd = 'chown {}:{} {}'.format(
+        vm.jailer.uid, vm.jailer.gid, jailed_handler)
+    run_cmd(cmd)
```

Comment:
wouldn't this be done by the removed `vm.create_jailed_resource(uffd_path)`?

Reply:
See comment above. Also, this snippet copies the page fault handler binary into the jailer, not the UDS.